components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -1661,25 +1661,80 @@ interface BraveWalletServiceTokenObserver {
   OnTokenRemoved(BlockchainToken token);
 };
 
-struct SignMessageRequest {
-  OriginInfo origin_info;
-  int32 id;
-  AccountId account_id;
-  string domain;
+// https://eips.ethereum.org/EIPS/eip-191
+struct EthStandardSignData {
+  // message for display
+  string message;
+};
+
+// https://eips.ethereum.org/EIPS/eip-712
+struct EthSignTypedData {
+  // message for display
   string message;
-  bool is_eip712;
-  // These fields are for hardware eip712 signing and
-  // Should be used only if is_eip712 is true
+  string domain;
+  // These fields are for hardware eip712 signing
   string? domain_hash;
   string? primary_hash;
+};
+
+// https://eips.ethereum.org/EIPS/eip-4361
+struct SIWEMessage {
+  url.mojom.Origin origin;
```

Comment:
Do we need this to be exactly `url.mojom.Origin` ?.
Afaik there are issues with this on UI platforms and in case origin is only needed for visualization we have OriginInfo for it(or maybe even string will suffice)